### PR TITLE
reCAPTCHA in sign-up page #1344

### LINKED ISSUE #1344: Add CAPTCHA in sign up page
**Expected Behaviour**

Add captcha in user sign up page

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1344 

Added google reCAPTCHA v2 with 2 options, audio and image.
Note: The captcha challenges wont resize if we resize the page due to material ui dialog box. Its not a problem from the captcha side. We can either make a new route for sign up page, without a dialog box after this.

**Changes:** [Add here what changes were made in this issue and if possible provide links.]

**Demo Link:** https://pr-1344-fossasia-susi-web-chat.surge.sh

**Screenshots for the change:** 

screen shot 2018-06-24 at 1 15 18 am
screen shot 2018-06-24 at 1 14 58 am

### CODE REVIEW

File: src/components/Auth/SignUp/SignUp.react.js [hunk 1]
```diff
@@ -405,7 +432,26 @@ export default class SignUp extends Component {
...
               />
             </div>
-
+            <div
+              style={{
+                marginTop: '10px',
+              }}
+            >
+              <Recaptcha
+                sitekey="6LfPZGAUAAAAANY4ZLR_lSat_YCuqqcV3sl84ulW"
```
Comment (Orbiter) [inline src/components/Auth/SignUp/SignUp.react.js]: To be able to maintain this we should have such kind of keys in a configuration file.

Comment (fragm3) [reply src/components/Auth/SignUp/SignUp.react.js]: I thought of this before, how and where should I put the key? @Orbiter

**Comment by fragm3** [comment]
@imjog I forgot to add the pr website in reCAPTCHA website, 
Check now it should work. url: https://pr-1344-fossasia-susi-web-chat.surge.sh/

File: src/components/Auth/SignUp/SignUp.react.js (hunk 1 above)
Comment (mariobehling) [reply src/components/Auth/SignUp/SignUp.react.js]: Add a config file into the root directory of the repo and make Recaptcha use the key from the config file.

**Comment by fragm3** [comment]
@PrP-11 Try multiple times, thats how captcha work.

**Comment by fragm3** [comment]
Added config file, Demo link: [physical-wine.surge.sh](http://www.physical-wine.surge.sh/)

**Comment by fragm3** [comment]
Please ignore the codacy erorr(double quotes)

File: src/components/Auth/SignUp/SignUp.react.js [hunk 2]
```diff
@@ -39,12 +42,29 @@ export default class SignUp extends Component {
     this.passwordErrorMessage = '';
     this.passwordConfirmErrorMessage = '';
     this.customServerMessage = '';
+    this.captchaverifyErrorMessage = '';
```
[the remark this answers is not in the document]
Comment (akshatnitd) [reply src/components/Auth/SignUp/SignUp.react.js]: Make it `captchaVerifyErrorMessage`.

[the remark this answers is not in the document]
Comment (akshatnitd) [reply src/components/Auth/SignUp/SignUp.react.js]: Also, squash the commits.

**Comment by fragm3** [comment]
Squashed the commits, please review @akshatnitd @mariobehling

**Comment by fragm3** [comment]
Codacy is failing due to double quotes, Travis CI error is not from this PR, but from older code.

**Comment by fragm3** [comment]
@Akshat-Jain please see previous comments :)

**Pull request merged**